doc/ERROR_CONCEPT.md

```diff
@@ -0,0 +1,250 @@
+* Permission/ Resource/ Parse/ External(C++, Ruby, Java)/.../ Miscellaneous:
+Most errors are resource errors (could not stat, no connection, could not close/open/find etc.) This again would not give users
+a particularly useful error indication since he/she has to lookup in the reason anyway. Even subdividing the `Resource` category
+would lead to the same problem. No matter how I look at it, the user
+will have to read the concrete reason in most cases which makes the Description line not that valuable any longer.
+* temporary/permanent/conflict/validation:
+This seems to be a better approach because it includes a possible solution for the user. How to categorize certain errors though
+seems impossible because there can be multiple reasons where to categorize an error. Take `Could not open file` as an example:
+if it is on a NFS and it is offline, *temporary* would be correct. If the file has too restrictive permissions, *permanent* would be more
+appropriate since the admin has to fix it. If the file is currently in use and semlocked by the system, *conflict* would be appropriate.
+* core kdb error/plugin error:
+Not much value for the user
+* differentiation to core features of elektra: Specification/ Notification/ KDB Core/ Bindings/ ...
+An error is again difficult to put into one bucket since some errors might be affected by multiple features (could not stat file for example
+which is used by multiple features).
+This approach would lead to the same problem as we have now (many errors, difficult to maintain)
+
+Personally I do not see how the benefit of a categorization outweighs the drawbacks because in by far the most cases users will have to look
+into the reason in more detail anyway. It should be enough for users to deal with the error just by the reason.
+
+As a result I would suggest to remove the description line.
+
+### Line 2: Error code
+
+Reasons to keep it:
+* Probably could be used to categorize errors such it is done in the http protocol
+* Good to use for shellrecorder scripts to test for errors
+* Possibility for automation (eg. if a certain error number occurred you could automatically trigger certain actions)
+
+Reasons against it:
+* Extra line
+
+Suggestion:
+Giving an error a number like it is done in C programs as return code seems to be outdated for elektra, especially when
+the reason is given anyway. Personally I never read the error number because it has not much value.
+The *reasons to keep* will also be very limited when the number of errors are significantly reduced.
+I would also suggest to remove it.
+
+### Line 1: General sentence
+
+All in all this sentence can stay because it gives the user a first hint on what happened.
+I would though change the wording and include more relevant information in it.
+I would change it like this:
+"Module <plugin> issued a <error/warning/fatal> while accessing the key database with the info:"
```

Comment:
Why having fatal? What is the difference?

Reply:
There are 7 fatal messages in the specification. I do not know their right to exist in detail but it looks like those cause elektra to malfunction or useless (failed to open backend, could not initialize modules)

Some errors though do not seem to be worth a fatal like "could not compile regex" (you actually commited this 8 years ago)

Reply:
Yes, in retrospective I think the distinction is arbitrary. I would remove this severity.